models/azure_project.rb

```diff
@@ -413,6 +423,39 @@ def get_compute_costs(cost_entries, date, rerun)
     compute_cost_log
   end
 
+  # This includes any resources tagged as core, including storage. Does not include data out costs.
+  def get_core_costs(cost_entries, date, rerun)
+    core_cost_log = self.cost_logs.find_by(date: date.to_s, scope: "core")
+
+    if !core_cost_log || rerun
+      core_costs = cost_entries.select do |cost|
+        cost["tags"] && cost["tags"]["type"] == "core" &&
+        !(cost['properties']["additionalInfo"] &&
+        JSON.parse(cost["properties"]["additionalInfo"])["UsageResourceKind"]&.include?("DataTrOut"))
+      end  
+      core_cost = begin
+                      core_costs.map { |c| c['properties']['cost'] }.reduce(:+)
+                     rescue NoMethodError
+                      0.0
+                     end
+      core_cost ||= 0.0
```

Comment:
When would this be necessary?

Reply:
Without this the `core_cost` is `nil` if there are no matched costs (`core_costs` is empty). It doesn't cause a problem at the moment as `nil.to_f` returns `0.0` but would be cleaner/safer to store the number.

Reply:
If `core_costs` is empty, wouldn't the `reduce` call raise `NoMethodError`?